docs/en/UI/Angular/HTTP-Error-Handling.md

```diff
@@ -0,0 +1,195 @@
+import { HTTP_ERROR_HANDLER, ... } from '@abp/ng.theme.shared';
+
+@NgModule({
+  // ...
+  providers: [
+    // ...
+    { provide: HTTP_ERROR_HANDLER, useValue: handleHttpErrors }
+  ],
+  declarations: [
+   //...
+   Error404Component],
+})
+export class AppModule {}
+```
+
+In the example above:
+
+- Created a function named `handleHttpErrors` and defined as value of the `HTTP_ERROR_HANDLER` provider in app.module. After this, the function executes when an HTTP error occurs.
+- 400 bad request errors is handled. When a 400 error occurs.
+
+- Since `of(httpError)` is returned at bottom of the `handleHttpErrors`, the `ErrorHandler` will handle the HTTP errors except 400 and 404 errors.
+
+**Note 1:** If you put `return EMPTY` to next line of handling an error, default error handling will not work for that error. [EMPTY](https://rxjs.dev/api/index/const/EMPTY) can be imported from `rxjs`.
+
+```ts
+export function handleHttpErrors(
+  injector: Injector,
+  httpError: HttpErrorResponse
+) {
+  if (httpError.status === 403) {
+    // handle 403 errors here
+    return EMPTY; // put return EMPTY to skip default error handling
+  }
+}
+```
+
+**Note 2:** If you put `return of(httpError)`, default error handling will work.
+
+- `of` is a function. It can be imported from `rxjs`.
+- `httpError` is the second parameter of the error handler function which is registered to the `HTTP_ERROR_HANDLER` provider. Type of the `httpError` is `HttpErrorResponse`.
+
+```ts
+import { of } from "rxjs";
+
+export function handleHttpErrors(
+  injector: Injector,
+  httpError: HttpErrorResponse
+) {
+  if (httpError.status === 500) {
+    // handle 500 errors here
+  }
+
+  // you can return the of(httpError) at bottom of the function to run the default handler of ABP for HTTP errors that you didn't handle above.
+  return of(httpError);
+}
+```
+
+### Service Method;
+
+With Services you can provide **more than one handler**.
+
+A custom HTTP error handler service can be registered to an injection token named **`CUSTOM_ERROR_HANDLERS`**.
+
+ABP has some default error handlers, you can see them [here](https://github.com/abpframework/abp/blob/dev/npm/ng-packs/packages/theme-shared/src/lib/providers/error-handlers.provider.ts).
+
+### How To Add New Handler Service?
+
+- ABP error handler services are implements the interface of **CustomHttpErrorHandlerService**.
+
+**Interface of `CUSTOM_ERROR_HANDLERS`**
+
+```ts
+interface CustomHttpErrorHandlerService {
+  readonly priority: number;
+  canHandle(error: unknown): boolean;
+  execute(): void;
+}
+```
+
+- **`priority`** ABP sorts the services according to the number of the priority variable. Higher priority will be checked first. You can think this as a z-index property in CSS.
+- **`canHandle(error: unknown): boolean :`** Check if the service can handle the error. Returns boolean.
+- **`execute(): void :`** If the service can handle the error, then run the execute method.

```

Comment:
- **`execute`** If the service can handle the error, then run the execute method